cmd/queue/main.go

```diff
@@ -272,25 +272,28 @@ func main() {
 		http.HandlerFunc(handler),
 	)
 
-	// Add a SIGTERM handler to gracefully shutdown the servers during
-	// pod termination.
+	go server.ListenAndServe()
+	go setupAdminHandlers(adminServer)
+
+	// Shutdown logic and signal handling
 	sigTermChan := make(chan os.Signal)
 	signal.Notify(sigTermChan, syscall.SIGTERM)
-	go func() {
-		<-sigTermChan
-		// Calling server.Shutdown() allows pending requests to
-		// complete, while no new work is accepted.
-
-		server.Shutdown(context.Background())
-		adminServer.Shutdown(context.Background())
+	// Blocks until we actually receive a TERM signal.
+	<-sigTermChan
+	// Calling server.Shutdown() allows pending requests to
+	// complete, while no new work is accepted.
+	logger.Debug("Received shutdown signal, attempting to gracefully shutdown servers.")
+	if err := server.Shutdown(context.Background()); err != nil {
+		logger.Error("Failed to shutdown proxy-server", zap.Error(err))
+	}
+	logger.Debug("Proxy server shutdown successfully.")
```

Comment:
a small comment, if the error happens above and we are on the debug level, both log messages would be discrepant

Reply:
Ha, good catch! Fixed!